app/uk/gov/hmrc/gform/models/FormModel.scala

```diff
@@ -234,6 +236,14 @@ case class FormModel[A <: PageMode](
         case head :: _ => availableSectionNumbers.reverse.find(_ < sectionNumber).getOrElse(head)
       }
     }
+  def maybeSectionNumbersFrom(fcId: FormComponentId): List[SectionNumber] =
```

Comment:
Is this return type lying? How can `fcid` be in multiple pages?

Reply:
I refactored as you suggested in the next comment with dedicated lookup 
`  val sectionNumberLookup: Map[FormComponentId, SectionNumber] = `